06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js

```diff
@@ -0,0 +1,141 @@
+    expect(invertedIndex.searchIndex('soon', 'work.json')).toEqual({
+      soon: [1],
+    });
+  });
+  it('should return true if the constant is a string', () => {
+    const words = 'Alice in Wonderland';
+    expect(Object.keys(invertedIndex.searchIndex(words))).toBeTruthy();
+  });
+  it('Should return correct index of the search term', () => {
+    expect(invertedIndex.searchIndex('humans', 'test.json')).toEqual({
+      humans: [1],
+    });
+  });
+  it('should return tokenized words for string in variable words', () => {
+    let words = 'It CAN only BE++ $$$$$   GOD';
+    const termsExpected = 'it can only be god';
+    words = invertedIndex.tokenize(words);
+    expect(termsExpected).toEqual(words);
+  });
+});
+
+
+describe('InvertedIndex : validateFile', () => {
+  it('should return truthy if book data is valid',
+    () => {
+      expect(Index.validateFile(workBook).status).toBeTruthy();
+    });
+  it('should return validate.msg if file uploaded is valid', () => {
+    expect(Index.validateFile(workBook).msg).toBe('This is a valid File');
+  });
+  it('should return falsy if book is not an array of object literals', () => {
+    expect(Index.validateFile('theDojo').status).toBeFalsy();
+    expect(Index.validateFile(100).status).toBeFalsy();
+  });
+  it('should return validate.msg if file uploaded is empty', () => {
+    expect(Index.validateFile('Dojo').msg).toBe('File is empty upload a file');
+    expect(Index.validateFile(100).msg).toBe('File is empty upload a file');
+  });
+  it('should return validate.msg if file dosent contain title or text', () => {
+    expect(Index.validateFile(empty).msg).toBe('Invalid file content');
+  });
+});
+
+describe('InvertedIndex : check type of InvertedIndex methods', () => {
+  it('should check that the class has a createIndex method', () => {
+    expect(typeof invertedIndex.createIndex).toBe('function');
+  });
+
+  it('should check that the class has a validateFile method', () => {
+    expect(typeof Index.validateFile).toBe('function');
+  });
+
+  it('should check that the class has a tokenize method', () => {
+    expect(typeof invertedIndex.tokenize).toBe('function');
+  });
+
+  it('should check that the class has a getIndex method', () => {
+    expect(typeof invertedIndex.getIndex).toBe('function');
+  });
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.searchIndex).toBe('function');
+  });
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.flattenSearch).toBe('function');
+  });
+});
+
+describe('InvertedIndex : Read file data ', () => {
+  it('should read and return the contents of a file via callback',
+   (read) => {
+     const file = new File('./files/test.json');
+     Index.readFile(file, (e) => {
+       expect(JSON.parse(e.target.result)).toEqual(test);
+       read();
+     });
+   });
+});
```

Comment:
Your test are not properly done. Repetition of test suites and specifications. Please review it